pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -53,6 +55,14 @@ while [ "$#" -gt 0 ]; do
       --rollback)
         rollback=1
         ;;
+      --generation)
+        if [ $# -eq 1 ]; then
```

Comment:
This should be
```sh
if [ $# -eq 0 ]; then
```
The code above (where `i` is assigned) shifts the index before you enter this code block.

Reply:
Good point...